woocommerce/woocommerce-ios
# Update local notifications to remove login experiments and prepare for store creation experiments

Part of #9618 
Prerequisite of #9665 


## Why


As part of #9618 to remove 2022 Q3 experiments, we're removing the login error notifications after product confirmation p1683190296583049/1682647903.364049-slack-C03L1NF1EA3. Then for the new experiments in #9665, we're adding 4 local notifications for store creation.

## How

This PR removed the feature flag `loginErrorNotifications` and added a new one `storeCreationNotifications` that is only enabled for debug builds. Then, we keep the same logic to request local notifications with provisional auth ([automatic trial of push notifications](https://developer.apple.com/documentation/usernotifications/asking_permission_to_use_notifications)) if the feature flag is enabled, just like before for the login error notifications. Then, 4 `LocalNotification` cases were added for the store creation experiments.

### Backward compatibility

Since there are still a few hundred local notifications scheduled every day according to Tracks (`woocommerceios_login_local_notification_scheduled`), the login error notification enums are kept and then used to cancel all of them so that we can remove any previously scheduled local notifications for pre-existing users.

## Testing instructions


The testing steps are more complicated, feel free to skip them as I tested it from two branches (`trunk` and PR branch).

- From `trunk`, update the login error notification time to be like 60 seconds away in `AuthenticationManager` below the comment `// 24 hours from now.`. Then launch the app
- Log out if needed
- Tap `Get started`, then `Log in`
- Enter some invalid email, then continue --> a local notification should be scheduled (feel free to set the time period to be lower to verify a local notification can be sent)
- Put the app to the background
- From the PR branch, quickly launch the app before the notification appears --> no local notifications should be sent

---
